SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -898,8 +898,12 @@ void CDiscAdjMultizoneDriver::ComputeAdjoints(unsigned short iZone, bool eval_tr
    *    on the last inner iteration. Structural problems have some minor issue and we
    *    need to evaluate this section on every iteration. ---*/
```

Comment:
Can someone comment on the "minor issue" that prevented skipping parts of the evaluation already when using a linear tape?

Reply:
Right, that part of the tape is skipped for inner iterations because it is not needed but it is expensive when there is mesh deformation (much more expensive that one fluid iteration).
However... Doing that skipping for structural problems results in a spike of the residuals on the last iteration (where the transfer section is evaluated). The derivatives do not change but the change in residuals was annoying to monitor convergence.
Because the structural problem does not have mesh deformation, it was "ok" to always evaluate the transfer to avoid that increase in residuals :shrug:

Reply:
Then this performance optimization should be kept. While the change fixes the `disc_adj_fsi` test case, it does not close all gaps in tape evaluations and the principle problem remains. For now, I will `assert` that index AD types do not support multiple zones.